spec.md

```diff
@@ -2369,6 +2369,85 @@ As an example `firetruck$y$z`{.firrtl} shares a prefix with
 `firetruck$y`{.firrtl} and `firetruck`{.firrtl}, but does not share a prefix
 with `fire`{.firrtl}.
 
+# Annotations
+
+Annotations encode arbitrary metadata and associate it with zero or more
+targets in a FIRRTL circuit.
+
+Annotations are represented as a dictionary, with a "class" field which
+describes which annotation it is, and a "target" field which represents the IR
+object it is attached to. The annotation's class matches the name of a Java
+class in the Scala Chisel/FIRRTL code base. Annotations may have arbitrary
+additional fields attached. Some annotation classes extend other annotations,
+which effectively means that the subclass annotation implies to effect of the
+parent annotation.
+
+Annotations are serializable to JSON and either live in a separate file
```

Comment:
We should include the format from @jackkoenig / @seldridge for putting annotations within a .fir file

Reply:
We were discussing on Slack... apparently this does not exist yet 😬

Reply:
I think the plan was to land this change first, then add the syntax for the separate `%[<jsonArray>]` syntax. (That would be a trivial follow-up.)